api/client/service/update.go

```diff
@@ -357,7 +357,12 @@ func removeItems(
 func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) {
 	if flags.Changed(flagMountAdd) {
 		values := flags.Lookup(flagMountAdd).Value.(*MountOpt).Value()
-		*mounts = append(*mounts, values...)
+
+		exist := findAndReplaceMountData(mounts, values)
```

Comment:
Comments are not clear to me.
- Since here, volume with duplicate target will be updated, so raising error is not required.
- Keeping mount ordering with updates make sense, will make changes for same.